src/task-controller.js

```diff
@@ -17,30 +17,61 @@
 import {SCHEDULER_PRIORITIES} from './scheduler-priorities.js';
 
 /**
- * Makes the controller's signal a TaskSignal by adding a read-only priority
- * property.
+ * The TaskSignal interface represents a signal object that allows you to
+ * communicate with a prioritized task, and abort it or change the priority
+ * via a TaskController object.
+ */
+class TaskSignal extends AbortSignal {
+  /**
+   * The priority of the task, user-visible by default.
+   * @readonly
+   * @type {string}
+   */
+  get priority() {
+    return 'user-visible';
+  }
+
+  /**
+   * The callback to be called when the priority of the task changes.
+   * @param {Function} callback
+   */
+  set onprioritychange(callback) {
+    if (this.onprioritychange_) {
+      this.removeEventListener('prioritychange', this.onprioritychange_);
+    }
+    this.addEventListener('prioritychange', callback);
+    /**
+     * @private
+     * @type {Function}
+     */
+    this.onprioritychange_ = callback;
+  }
+
+  /**
+   * The callback to be called when the priority of the task changes.
+   * @type {Function}
+   */
+  get onprioritychange() {
+    return this.onprioritychange_ || null;
+  }
+}
+
+/**
+ * Makes the TaskSignal instance from the AbortController instance.
  * @private
  * @param {TaskController} controller
  */
 function makeTaskSignal(controller) {
-  const signal = controller.signal;
-  Object.defineProperties(signal, {
-    priority: {
-      get: function() {
-        return controller.priority_;
-      },
-      enumerable: true,
-    },
-    onprioritychange: {
-      value: null,
-      writable: true,
-      enumerable: true,
+  // Change the prototype of the signal to TaskSignal.prototype.
+  Object.setPrototypeOf(controller.signal, TaskSignal.prototype);
+
+  // Connect the priority property to the controller's priority.
+  Object.defineProperty(controller.signal, 'priority', {
+    get: function priority() {
+      return controller.priority_;
     },
-  });
-  signal.addEventListener('prioritychange', (e) => {
-    if (signal.onprioritychange) {
-      signal.onprioritychange(e);
-    }
+    enumerable: true,
+    configurable: true,
```

Comment:
This is how it's defined in native TaskSignal:
<img width="493" alt="image" src="https://github.com/user-attachments/assets/7165dc11-d8d0-4f53-ba6d-cfe907d1b4f8" />